provider/src/main/java/com/cloudera/director/google/sql/GoogleCloudSQLInstanceTemplateConfigurationValidator.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2015 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cloudera.director.google.sql;
+
+import static com.cloudera.director.google.sql.GoogleCloudSQLInstanceTemplateConfigurationProperty.TIER;
+import static com.cloudera.director.spi.v1.model.InstanceTemplate.InstanceTemplateConfigurationPropertyToken.INSTANCE_NAME_PREFIX;
+import static com.cloudera.director.spi.v1.model.util.Validations.addError;
+
+import com.cloudera.director.google.internal.GoogleCredentials;
+import com.cloudera.director.spi.v1.model.ConfigurationValidator;
+import com.cloudera.director.spi.v1.model.Configured;
+import com.cloudera.director.spi.v1.model.LocalizationContext;
+import com.cloudera.director.spi.v1.model.exception.PluginExceptionConditionAccumulator;
+import com.cloudera.director.spi.v1.model.exception.TransientProviderException;
+import com.google.api.services.sqladmin.model.Tier;
+import com.google.api.services.sqladmin.SQLAdmin;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class GoogleCloudSQLInstanceTemplateConfigurationValidator implements ConfigurationValidator  {
+  private static final Logger LOG =
+      LoggerFactory.getLogger(GoogleCloudSQLInstanceTemplateConfigurationValidator.class);
+
+  @VisibleForTesting
+  static final String INVALID_TIER_MSG = "Tier '%s' not found.";
```

Comment:
Does Cloudera Director do internationalization / localization for user-visible messages? Should we be doing something to make it easy to translate these to new languages in the future?

/cc: @andreisavu, @jadair-cloudera for comment